solr/core/src/java/org/apache/solr/cloud/overseer/ReplicaMutator.java

```diff
@@ -345,6 +346,14 @@ private ZkWriteCommand updateState(final ClusterState prevState, ZkNodeProps mes
     Map<String, Replica> replicas;
 
     if (slice != null) {
+      if (coreNodeName != null) {
+        if (isModifiedReplica(slice.getReplica(coreNodeName), message)) {
+          //one of the basic properties of the replica got changed such ar core/base_url/ node_name
```

Comment:
When/how core name can change?

Reply:
CCing @noblepaul @chatman

Reply:
port is ok but why corename?

Reply:
I won't do this for PRS.

Reply:
This way we don't need to update PRS state - just update the state.json file for port/http/s change